src/nvim/mbyte.c

```diff
@@ -1048,11 +1054,13 @@ bool utf_printable(int c)
 {
   // Sorted list of non-overlapping intervals.
   // 0xd800-0xdfff is reserved for UTF-16, actually illegal.
+  // 0x10FFFF is the last valid codepoints, all values above that
+  // must be escaped.
   static struct interval nonprint[] =
   {
     { 0x070f, 0x070f }, { 0x180b, 0x180e }, { 0x200b, 0x200f }, { 0x202a, 0x202e },
     { 0x2060, 0x206f }, { 0xd800, 0xdfff }, { 0xfeff, 0xfeff }, { 0xfff9, 0xfffb },
-    { 0xfffe, 0xffff }
+    { 0xfffe, 0xffff }, { 0x110000, 0x8fffffff },
```

Comment:
`interval` is two `long`s, on platforms with 32-bit `long` it is limited to `0x7fffffff`, so `0x8fffffff` will overflow and become a negative number?

Reply:
sure, should have been `7f`